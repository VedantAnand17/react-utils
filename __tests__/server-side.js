import mockdate from 'mockdate';

mockdate.set('2020-04-19Z');

test('Export at server side', () => {
  const lib = require('../src');
  expect(lib).toMatchSnapshot();
  expect({ ...lib._ }).toMatchSnapshot();
  expect({ ...lib.api }).toMatchSnapshot();
  expect(Object.keys(lib.time)).toMatchSnapshot();

  Object.entries(lib.server).forEach((item) => expect(item).toMatchSnapshot());
});
