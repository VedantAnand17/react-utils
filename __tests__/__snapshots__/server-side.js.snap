// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export at server side 1`] = `
{
  "Barrier": [Function],
  "BaseModal": [Function],
  "Button": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
    "themeType": [Function],
  },
  "Checkbox": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
    "themeType": [Function],
  },
  "Dropdown": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
    "themeType": [Function],
  },
  "Emitter": [Function],
  "GlobalStateProvider": [Function],
  "Input": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
    "themeType": [Function],
  },
  "JU": {
    "act": [Function],
    "getMockUuid": [Function],
    "mockClientSide": [Function],
    "mockTimer": [Function],
    "mount": [Function],
    "render": [Function],
    "shallowRender": [Function],
    "shallowSnapshot": [Function],
    "simulate": {
      "abort": [Function],
      "animationEnd": [Function],
      "animationIteration": [Function],
      "animationStart": [Function],
      "auxClick": [Function],
      "beforeInput": [Function],
      "blur": [Function],
      "canPlay": [Function],
      "canPlayThrough": [Function],
      "cancel": [Function],
      "change": [Function],
      "click": [Function],
      "close": [Function],
      "compositionEnd": [Function],
      "compositionStart": [Function],
      "compositionUpdate": [Function],
      "contextMenu": [Function],
      "copy": [Function],
      "cut": [Function],
      "doubleClick": [Function],
      "drag": [Function],
      "dragEnd": [Function],
      "dragEnter": [Function],
      "dragExit": [Function],
      "dragLeave": [Function],
      "dragOver": [Function],
      "dragStart": [Function],
      "drop": [Function],
      "durationChange": [Function],
      "emptied": [Function],
      "encrypted": [Function],
      "ended": [Function],
      "error": [Function],
      "focus": [Function],
      "gotPointerCapture": [Function],
      "input": [Function],
      "invalid": [Function],
      "keyDown": [Function],
      "keyPress": [Function],
      "keyUp": [Function],
      "load": [Function],
      "loadStart": [Function],
      "loadedData": [Function],
      "loadedMetadata": [Function],
      "lostPointerCapture": [Function],
      "mouseDown": [Function],
      "mouseEnter": [Function],
      "mouseLeave": [Function],
      "mouseMove": [Function],
      "mouseOut": [Function],
      "mouseOver": [Function],
      "mouseUp": [Function],
      "paste": [Function],
      "pause": [Function],
      "play": [Function],
      "playing": [Function],
      "pointerCancel": [Function],
      "pointerDown": [Function],
      "pointerEnter": [Function],
      "pointerLeave": [Function],
      "pointerMove": [Function],
      "pointerOut": [Function],
      "pointerOver": [Function],
      "pointerUp": [Function],
      "progress": [Function],
      "rateChange": [Function],
      "reset": [Function],
      "resize": [Function],
      "scroll": [Function],
      "seeked": [Function],
      "seeking": [Function],
      "select": [Function],
      "stalled": [Function],
      "submit": [Function],
      "suspend": [Function],
      "timeUpdate": [Function],
      "toggle": [Function],
      "touchCancel": [Function],
      "touchEnd": [Function],
      "touchMove": [Function],
      "touchStart": [Function],
      "transitionEnd": [Function],
      "volumeChange": [Function],
      "waiting": [Function],
      "wheel": [Function],
    },
    "snapshot": [Function],
    "unmockClientSide": [Function],
  },
  "Link": [Function],
  "MetaTags": [Function],
  "Modal": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
    "themeType": [Function],
  },
  "NavLink": [Function],
  "PT": {
    "PropTypes": {
      "PropTypes": [Circular],
      "any": [Function],
      "array": [Function],
      "arrayOf": [Function],
      "bigint": [Function],
      "bool": [Function],
      "checkPropTypes": [Function],
      "element": [Function],
      "elementType": [Function],
      "exact": [Function],
      "func": [Function],
      "instanceOf": [Function],
      "node": [Function],
      "number": [Function],
      "object": [Function],
      "objectOf": [Function],
      "oneOf": [Function],
      "oneOfType": [Function],
      "resetWarningCache": [Function],
      "shape": [Function],
      "string": [Function],
      "symbol": [Function],
    },
    "any": [Function],
    "array": [Function],
    "arrayOf": [Function],
    "bigint": [Function],
    "bool": [Function],
    "checkPropTypes": [Function],
    "default": {
      "PropTypes": [Circular],
      "any": [Function],
      "array": [Function],
      "arrayOf": [Function],
      "bigint": [Function],
      "bool": [Function],
      "checkPropTypes": [Function],
      "element": [Function],
      "elementType": [Function],
      "exact": [Function],
      "func": [Function],
      "instanceOf": [Function],
      "node": [Function],
      "number": [Function],
      "object": [Function],
      "objectOf": [Function],
      "oneOf": [Function],
      "oneOfType": [Function],
      "resetWarningCache": [Function],
      "shape": [Function],
      "string": [Function],
      "symbol": [Function],
    },
    "element": [Function],
    "elementType": [Function],
    "exact": [Function],
    "func": [Function],
    "instanceOf": [Function],
    "node": [Function],
    "number": [Function],
    "object": [Function],
    "objectOf": [Function],
    "oneOf": [Function],
    "oneOfType": [Function],
    "resetWarningCache": [Function],
    "shape": [Function],
    "string": [Function],
    "symbol": [Function],
  },
  "PageLayout": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
    "themeType": [Function],
  },
  "ScalableRect": [Function],
  "Semaphore": [Function],
  "ThemeProvider": [Function],
  "Throbber": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
    "themeType": [Function],
  },
  "WithTooltip": {
    "$$typeof": Symbol(react.forward_ref),
    "PLACEMENTS": {
      "ABOVE_CURSOR": "ABOVE_CURSOR",
      "ABOVE_ELEMENT": "ABOVE_ELEMENT",
      "BELOW_CURSOR": "BELOW_CURSOR",
      "BELOW_ELEMENT": "BELOW_ELEMENT",
    },
    "render": [Function],
    "themeType": [Function],
  },
  "YouTubeVideo": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
    "themeType": [Function],
  },
  "api": [Function],
  "client": [Function],
  "config": {
    "SECRET": {
      "dummySecretKey": "Dummy Secret Value",
    },
    "dummyKey": "Dummy Value",
  },
  "getGlobalState": [Function],
  "getSsrContext": [Function],
  "isomorphy": {
    "IS_CLIENT_SIDE": false,
    "IS_SERVER_SIDE": true,
    "buildTimestamp": [Function],
    "getBuildInfo": [Function],
    "isDevBuild": [Function],
    "isProdBuild": [Function],
  },
  "newBarrier": [Function],
  "server": [Function],
  "splitComponent": [Function],
  "themed": [Function],
  "time": [Function],
  "useAsyncCollection": [Function],
  "useAsyncData": [Function],
  "useGlobalState": [Function],
  "webpack": {
    "requireWeak": [Function],
    "resolveWeak": [Function],
  },
  "withRetries": [Function],
}
`;

exports[`Export at server side 2`] = `{}`;

exports[`Export at server side 3`] = `
wrap {
  "Axios": [Function],
  "AxiosError": [Function],
  "AxiosHeaders": [Function],
  "Cancel": [Function],
  "CancelToken": [Function],
  "CanceledError": [Function],
  "HttpStatusCode": {
    "100": "Continue",
    "101": "SwitchingProtocols",
    "102": "Processing",
    "103": "EarlyHints",
    "200": "Ok",
    "201": "Created",
    "202": "Accepted",
    "203": "NonAuthoritativeInformation",
    "204": "NoContent",
    "205": "ResetContent",
    "206": "PartialContent",
    "207": "MultiStatus",
    "208": "AlreadyReported",
    "226": "ImUsed",
    "300": "MultipleChoices",
    "301": "MovedPermanently",
    "302": "Found",
    "303": "SeeOther",
    "304": "NotModified",
    "305": "UseProxy",
    "306": "Unused",
    "307": "TemporaryRedirect",
    "308": "PermanentRedirect",
    "400": "BadRequest",
    "401": "Unauthorized",
    "402": "PaymentRequired",
    "403": "Forbidden",
    "404": "NotFound",
    "405": "MethodNotAllowed",
    "406": "NotAcceptable",
    "407": "ProxyAuthenticationRequired",
    "408": "RequestTimeout",
    "409": "Conflict",
    "410": "Gone",
    "411": "LengthRequired",
    "412": "PreconditionFailed",
    "413": "PayloadTooLarge",
    "414": "UriTooLong",
    "415": "UnsupportedMediaType",
    "416": "RangeNotSatisfiable",
    "417": "ExpectationFailed",
    "418": "ImATeapot",
    "421": "MisdirectedRequest",
    "422": "UnprocessableEntity",
    "423": "Locked",
    "424": "FailedDependency",
    "425": "TooEarly",
    "426": "UpgradeRequired",
    "428": "PreconditionRequired",
    "429": "TooManyRequests",
    "431": "RequestHeaderFieldsTooLarge",
    "451": "UnavailableForLegalReasons",
    "500": "InternalServerError",
    "501": "NotImplemented",
    "502": "BadGateway",
    "503": "ServiceUnavailable",
    "504": "GatewayTimeout",
    "505": "HttpVersionNotSupported",
    "506": "VariantAlsoNegotiates",
    "507": "InsufficientStorage",
    "508": "LoopDetected",
    "510": "NotExtended",
    "511": "NetworkAuthenticationRequired",
    "Accepted": 202,
    "AlreadyReported": 208,
    "BadGateway": 502,
    "BadRequest": 400,
    "Conflict": 409,
    "Continue": 100,
    "Created": 201,
    "EarlyHints": 103,
    "ExpectationFailed": 417,
    "FailedDependency": 424,
    "Forbidden": 403,
    "Found": 302,
    "GatewayTimeout": 504,
    "Gone": 410,
    "HttpVersionNotSupported": 505,
    "ImATeapot": 418,
    "ImUsed": 226,
    "InsufficientStorage": 507,
    "InternalServerError": 500,
    "LengthRequired": 411,
    "Locked": 423,
    "LoopDetected": 508,
    "MethodNotAllowed": 405,
    "MisdirectedRequest": 421,
    "MovedPermanently": 301,
    "MultiStatus": 207,
    "MultipleChoices": 300,
    "NetworkAuthenticationRequired": 511,
    "NoContent": 204,
    "NonAuthoritativeInformation": 203,
    "NotAcceptable": 406,
    "NotExtended": 510,
    "NotFound": 404,
    "NotImplemented": 501,
    "NotModified": 304,
    "Ok": 200,
    "PartialContent": 206,
    "PayloadTooLarge": 413,
    "PaymentRequired": 402,
    "PermanentRedirect": 308,
    "PreconditionFailed": 412,
    "PreconditionRequired": 428,
    "Processing": 102,
    "ProxyAuthenticationRequired": 407,
    "RangeNotSatisfiable": 416,
    "RequestHeaderFieldsTooLarge": 431,
    "RequestTimeout": 408,
    "ResetContent": 205,
    "SeeOther": 303,
    "ServiceUnavailable": 503,
    "SwitchingProtocols": 101,
    "TemporaryRedirect": 307,
    "TooEarly": 425,
    "TooManyRequests": 429,
    "Unauthorized": 401,
    "UnavailableForLegalReasons": 451,
    "UnprocessableEntity": 422,
    "UnsupportedMediaType": 415,
    "Unused": 306,
    "UpgradeRequired": 426,
    "UriTooLong": 414,
    "UseProxy": 305,
    "VariantAlsoNegotiates": 506,
  },
  "VERSION": "1.3.4",
  "all": [Function],
  "constructor": [Function],
  "create": [Function],
  "default": [Function],
  "defaults": {
    "adapter": [
      "xhr",
      "http",
    ],
    "env": {
      "Blob": [Function],
      "FormData": [Function],
    },
    "headers": {
      "common": {
        "Accept": "application/json, text/plain, */*",
      },
      "delete": {},
      "get": {},
      "head": {},
      "patch": {
        "Content-Type": undefined,
      },
      "post": {
        "Content-Type": undefined,
      },
      "put": {
        "Content-Type": undefined,
      },
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "timeout": 0,
    "transformRequest": [
      [Function],
    ],
    "transformResponse": [
      [Function],
    ],
    "transitional": {
      "clarifyTimeoutError": false,
      "forcedJSONParsing": true,
      "silentJSONParsing": true,
    },
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "delete": [Function],
  "formToJSON": [Function],
  "get": [Function],
  "getUri": [Function],
  "head": [Function],
  "interceptors": {
    "request": InterceptorManager {
      "handlers": [],
    },
    "response": InterceptorManager {
      "handlers": [],
    },
  },
  "isAxiosError": [Function],
  "isCancel": [Function],
  "mergeConfig": [Function],
  "options": [Function],
  "patch": [Function],
  "patchForm": [Function],
  "post": [Function],
  "postForm": [Function],
  "put": [Function],
  "putForm": [Function],
  "request": [Function],
  "spread": [Function],
  "toFormData": [Function],
}
`;

exports[`Export at server side 4`] = `
[
  "extend",
  "locale",
  "isDayjs",
  "unix",
  "en",
  "Ls",
  "p",
  "SEC_MS",
  "MIN_MS",
  "HOUR_MS",
  "DAY_MS",
  "YEAR_MS",
  "now",
  "timer",
  "useCurrent",
  "useTimezoneOffset",
]
`;

exports[`Export at server side 5`] = `
[
  "SCRIPT_LOCATIONS",
  {
    "BODY_OPEN": "BODY_OPEN",
    "DEFAULT": "DEFAULT",
    "HEAD_OPEN": "HEAD_OPEN",
  },
]
`;

exports[`Export at server side 6`] = `
[
  "getDefaultCspSettings",
  [Function],
]
`;

exports[`Export at server side 7`] = `
[
  "errors",
  {
    "CODES": {
      "100": "CONTINUE",
      "101": "SWITCHING_PROTOCOLS",
      "102": "PROCESSING",
      "200": "OK",
      "201": "CREATED",
      "202": "ACCEPTED",
      "203": "NON_AUTHORITATIVE_INFORMATION",
      "204": "NO_CONTENT",
      "205": "RESET_CONTENT",
      "206": "PARTIAL_CONTENT",
      "207": "MULTI_STATUS",
      "300": "MULTIPLE_CHOICES",
      "301": "MOVED_PERMANENTLY",
      "302": "MOVED_TEMPORARILY",
      "303": "SEE_OTHER",
      "304": "NOT_MODIFIED",
      "305": "USE_PROXY",
      "307": "TEMPORARY_REDIRECT",
      "308": "PERMANENT_REDIRECT",
      "400": "BAD_REQUEST",
      "401": "UNAUTHORIZED",
      "402": "PAYMENT_REQUIRED",
      "403": "FORBIDDEN",
      "404": "NOT_FOUND",
      "405": "METHOD_NOT_ALLOWED",
      "406": "NOT_ACCEPTABLE",
      "407": "PROXY_AUTHENTICATION_REQUIRED",
      "408": "REQUEST_TIMEOUT",
      "409": "CONFLICT",
      "410": "GONE",
      "411": "LENGTH_REQUIRED",
      "412": "PRECONDITION_FAILED",
      "413": "REQUEST_TOO_LONG",
      "414": "REQUEST_URI_TOO_LONG",
      "415": "UNSUPPORTED_MEDIA_TYPE",
      "416": "REQUESTED_RANGE_NOT_SATISFIABLE",
      "417": "EXPECTATION_FAILED",
      "418": "IM_A_TEAPOT",
      "419": "INSUFFICIENT_SPACE_ON_RESOURCE",
      "420": "METHOD_FAILURE",
      "421": "MISDIRECTED_REQUEST",
      "422": "UNPROCESSABLE_ENTITY",
      "423": "LOCKED",
      "424": "FAILED_DEPENDENCY",
      "428": "PRECONDITION_REQUIRED",
      "429": "TOO_MANY_REQUESTS",
      "431": "REQUEST_HEADER_FIELDS_TOO_LARGE",
      "451": "UNAVAILABLE_FOR_LEGAL_REASONS",
      "500": "INTERNAL_SERVER_ERROR",
      "501": "NOT_IMPLEMENTED",
      "502": "BAD_GATEWAY",
      "503": "SERVICE_UNAVAILABLE",
      "504": "GATEWAY_TIMEOUT",
      "505": "HTTP_VERSION_NOT_SUPPORTED",
      "507": "INSUFFICIENT_STORAGE",
      "511": "NETWORK_AUTHENTICATION_REQUIRED",
      "ACCEPTED": 202,
      "BAD_GATEWAY": 502,
      "BAD_REQUEST": 400,
      "CONFLICT": 409,
      "CONTINUE": 100,
      "CREATED": 201,
      "EXPECTATION_FAILED": 417,
      "FAILED_DEPENDENCY": 424,
      "FORBIDDEN": 403,
      "GATEWAY_TIMEOUT": 504,
      "GONE": 410,
      "HTTP_VERSION_NOT_SUPPORTED": 505,
      "IM_A_TEAPOT": 418,
      "INSUFFICIENT_SPACE_ON_RESOURCE": 419,
      "INSUFFICIENT_STORAGE": 507,
      "INTERNAL_SERVER_ERROR": 500,
      "LENGTH_REQUIRED": 411,
      "LOCKED": 423,
      "METHOD_FAILURE": 420,
      "METHOD_NOT_ALLOWED": 405,
      "MISDIRECTED_REQUEST": 421,
      "MOVED_PERMANENTLY": 301,
      "MOVED_TEMPORARILY": 302,
      "MULTIPLE_CHOICES": 300,
      "MULTI_STATUS": 207,
      "NETWORK_AUTHENTICATION_REQUIRED": 511,
      "NON_AUTHORITATIVE_INFORMATION": 203,
      "NOT_ACCEPTABLE": 406,
      "NOT_FOUND": 404,
      "NOT_IMPLEMENTED": 501,
      "NOT_MODIFIED": 304,
      "NO_CONTENT": 204,
      "OK": 200,
      "PARTIAL_CONTENT": 206,
      "PAYMENT_REQUIRED": 402,
      "PERMANENT_REDIRECT": 308,
      "PRECONDITION_FAILED": 412,
      "PRECONDITION_REQUIRED": 428,
      "PROCESSING": 102,
      "PROXY_AUTHENTICATION_REQUIRED": 407,
      "REQUESTED_RANGE_NOT_SATISFIABLE": 416,
      "REQUEST_HEADER_FIELDS_TOO_LARGE": 431,
      "REQUEST_TIMEOUT": 408,
      "REQUEST_TOO_LONG": 413,
      "REQUEST_URI_TOO_LONG": 414,
      "RESET_CONTENT": 205,
      "SEE_OTHER": 303,
      "SERVICE_UNAVAILABLE": 503,
      "SWITCHING_PROTOCOLS": 101,
      "TEMPORARY_REDIRECT": 307,
      "TOO_MANY_REQUESTS": 429,
      "UNAUTHORIZED": 401,
      "UNAVAILABLE_FOR_LEGAL_REASONS": 451,
      "UNPROCESSABLE_ENTITY": 422,
      "UNSUPPORTED_MEDIA_TYPE": 415,
      "USE_PROXY": 305,
    },
    "ERRORS": {
      "ACCEPTED": "Accepted",
      "BAD_GATEWAY": "Bad Gateway",
      "BAD_REQUEST": "Bad Request",
      "CONFLICT": "Conflict",
      "CONTINUE": "Continue",
      "CREATED": "Created",
      "EXPECTATION_FAILED": "Expectation Failed",
      "FAILED_DEPENDENCY": "Failed Dependency",
      "FORBIDDEN": "Forbidden",
      "GATEWAY_TIMEOUT": "Gateway Timeout",
      "GONE": "Gone",
      "HTTP_VERSION_NOT_SUPPORTED": "HTTP Version Not Supported",
      "IM_A_TEAPOT": "I'm a teapot",
      "INSUFFICIENT_SPACE_ON_RESOURCE": "Insufficient Space on Resource",
      "INSUFFICIENT_STORAGE": "Insufficient Storage",
      "INTERNAL_SERVER_ERROR": "Internal Server Error",
      "LENGTH_REQUIRED": "Length Required",
      "LOCKED": "Locked",
      "METHOD_FAILURE": "Method Failure",
      "METHOD_NOT_ALLOWED": "Method Not Allowed",
      "MISDIRECTED_REQUEST": "Misdirected Request",
      "MOVED_PERMANENTLY": "Moved Permanently",
      "MOVED_TEMPORARILY": "Moved Temporarily",
      "MULTIPLE_CHOICES": "Multiple Choices",
      "MULTI_STATUS": "Multi-Status",
      "NETWORK_AUTHENTICATION_REQUIRED": "Network Authentication Required",
      "NON_AUTHORITATIVE_INFORMATION": "Non Authoritative Information",
      "NOT_ACCEPTABLE": "Not Acceptable",
      "NOT_FOUND": "Not Found",
      "NOT_IMPLEMENTED": "Not Implemented",
      "NOT_MODIFIED": "Not Modified",
      "NO_CONTENT": "No Content",
      "OK": "OK",
      "PARTIAL_CONTENT": "Partial Content",
      "PAYMENT_REQUIRED": "Payment Required",
      "PERMANENT_REDIRECT": "Permanent Redirect",
      "PRECONDITION_FAILED": "Precondition Failed",
      "PRECONDITION_REQUIRED": "Precondition Required",
      "PROCESSING": "Processing",
      "PROXY_AUTHENTICATION_REQUIRED": "Proxy Authentication Required",
      "REQUESTED_RANGE_NOT_SATISFIABLE": "Requested Range Not Satisfiable",
      "REQUEST_HEADER_FIELDS_TOO_LARGE": "Request Header Fields Too Large",
      "REQUEST_TIMEOUT": "Request Timeout",
      "REQUEST_TOO_LONG": "Request Entity Too Large",
      "REQUEST_URI_TOO_LONG": "Request-URI Too Long",
      "RESET_CONTENT": "Reset Content",
      "SEE_OTHER": "See Other",
      "SERVICE_UNAVAILABLE": "Service Unavailable",
      "SWITCHING_PROTOCOLS": "Switching Protocols",
      "TEMPORARY_REDIRECT": "Temporary Redirect",
      "TOO_MANY_REQUESTS": "Too Many Requests",
      "UNAUTHORIZED": "Unauthorized",
      "UNAVAILABLE_FOR_LEGAL_REASONS": "Unavailable For Legal Reasons",
      "UNPROCESSABLE_ENTITY": "Unprocessable Entity",
      "UNSUPPORTED_MEDIA_TYPE": "Unsupported Media Type",
      "USE_PROXY": "Use Proxy",
    },
    "assert": [Function],
    "fail": [Function],
    "getErrorForCode": [Function],
    "joi": {
      "ValidationError": [Function],
      "_types": Set {
        "alternatives",
        "any",
        "array",
        "boolean",
        "date",
        "function",
        "link",
        "number",
        "object",
        "string",
        "symbol",
        "binary",
      },
      "allow": [Function],
      "alt": [Function],
      "alternatives": [Function],
      "any": [Function],
      "array": [Function],
      "assert": [Function],
      "attempt": [Function],
      "binary": [Function],
      "bool": [Function],
      "boolean": [Function],
      "build": [Function],
      "cache": {
        "provision": [Function],
      },
      "checkPreferences": [Function],
      "compile": [Function],
      "custom": [Function],
      "date": [Function],
      "defaults": [Function],
      "disallow": [Function],
      "equal": [Function],
      "exist": [Function],
      "expression": [Function],
      "extend": [Function],
      "forbidden": [Function],
      "func": [Function],
      "function": [Function],
      "in": [Function],
      "invalid": [Function],
      "isError": [Function],
      "isExpression": [Function],
      "isRef": [Function],
      "isSchema": [Function],
      "link": [Function],
      "not": [Function],
      "number": [Function],
      "object": [Function],
      "only": [Function],
      "optional": [Function],
      "options": [Function],
      "override": Symbol(override),
      "preferences": [Function],
      "prefs": [Function],
      "ref": [Function],
      "required": [Function],
      "string": [Function],
      "strip": [Function],
      "symbol": [Function],
      "trace": [Function],
      "types": [Function],
      "untrace": [Function],
      "valid": [Function],
      "version": "17.9.1",
      "when": [Function],
      "x": [Function],
      Symbol(@hapi/lab/coverage/initialize): [Function],
    },
    "newError": [Function],
  },
]
`;
